src/lib/schematics/update/rules/switchIdentifiersRule.ts

```diff
@@ -0,0 +1,135 @@
+  }
+}
+
+export class SwitchIdentifiersWalker extends ProgramAwareRuleWalker {
+  constructor(sf, opt, prog) {
+    super(sf, opt, prog);
+  }
+
+  /** List of Angular Material declarations inside of the current source file. */
+  materialDeclarations: ts.Declaration[] = [];
+
+  /** List of Angular Material namespace declarations in the current source file. */
+  materialNamespaceDeclarations: ts.Declaration[] = [];
+
+  /** Method that is called for every identifier inside of the specified project. */
+  visitIdentifier(identifier: ts.Identifier) {
+    // Store Angular Material namespace identifers in a list of declarations.
+    // Namespace identifiers can be: `import * as md from '@angular/material';`
+    this._storeNamespaceImports(identifier);
+
+    // For identifiers that aren't listed in the className data, the whole check can be
+    // skipped safely.
+    if (!classNames.some(data => data.replace === identifier.text)) {
+      return;
+    }
+
+    const symbol = getOriginalSymbolFromNode(identifier, this.getTypeChecker());
+
+    // If the symbol is not defined or could not be resolved, just skip the following identifier
+    // checks.
+    if (!symbol || !symbol.name || symbol.name === 'unknown') {
+      console.error(`Could not resolve symbol for identifier "${identifier.text}" ` +
+          `in file ${this._getRelativeFileName()}`);
+      return;
+    }
+
+    // For export declarations that are referring to Angular Material, the identifier should be
+    // switched to the new name.
+    if (isExportSpecifierNode(identifier) && isMaterialExportDeclaration(identifier)) {
+      return this.createIdentifierFailure(identifier, symbol);
+    }
+
+    // For import declarations that are referring to Angular Material, the value declarations
+    // should be stored so that other identifiers in the file can be compared.
+    if (isImportSpecifierNode(identifier) && isMaterialImportDeclaration(identifier)) {
+      this.materialDeclarations.push(symbol.valueDeclaration);
+    }
+
+    // For identifiers that are not part of an import or export, the list of Material declarations
+    // should be checked to ensure that only identifiers of Angular Material are updated.
+    // Identifiers that are imported through an Angular Material namespace will be updated.
+    else if (this.materialDeclarations.indexOf(symbol.valueDeclaration) === -1 &&
+              !this._isIdentifierFromNamespace(identifier)) {
+      return;
+    }
+
+    return this.createIdentifierFailure(identifier, symbol);
+  }
+
+  /** Creates a failure and replacement for the specified identifier. */
+  private createIdentifierFailure(identifier: ts.Identifier, symbol: ts.Symbol) {
+    let classData = classNames.find(
+        data => data.replace === symbol.name || data.replace === identifier.text);
+
+    if (!classData) {
+      console.error(`Could not find updated name for identifier "${identifier.getText()}" in ` +
```

Comment:
Should we console.error or throw error here?

Reply:
`console.error`, we still want to continue trying to detect and fix other issues